docs/pubsub_snippets.py

```diff
@@ -0,0 +1,313 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Testable usage examples for Google Cloud Pubsub API wrapper
+
+Each example function takes a ``client`` argument (which must be an instance
+of :class:`gcloud.pubsub.client.Client`) and uses it to perform a task with
+the API.
+
+To facility running the examples as system tests, each example is also passed
+a ``to_delete`` list;  the function adds to the list any objects created which
+need to be deleted during teardown.
+"""
+
+import time
+
+from gcloud.pubsub.client import Client
+
+
+def snippet(func):
+    """Mark ``func`` as a snippet example function."""
+    func._snippet = True
+    return func
+
+def _millis():
+    return time.time() * 1000
+
+
+@snippet
+def client_list_topics(client, to_delete):
+    """List topics for a project."""
+    def do_something_with(sub):
+        pass
+
+    # [START client_list_topics]
```

Comment:
It doesn't de-dent, and I think it looks *better* that way:
![screenshot from 2016-04-01 16 18 47](https://cloud.githubusercontent.com/assets/242750/14218969/79fa71f6-f825-11e5-90c8-24da55190c24.png)